lib/matplotlib/backends/backend_webagg_core.py

```diff
@@ -241,8 +240,8 @@ def handle_event(self, event):
         return handler(event)
 
     def handle_unknown_event(self, event):
-        warnings.warn('Unhandled message type {0}. {1}'.format(
-            event['type'], event), stacklevel=2)
+        cbook._warn_external('Unhandled message type {0}. {1}'.format(
```

Comment:
I think this should use logging (it's basically an event handler and thus likely to be called asynchronously anyways).